ethereum/core/src/main/java/org/hyperledger/besu/ethereum/worldstate/Pruner.java

```diff
@@ -70,27 +68,27 @@ public Pruner(
       final MarkSweepPruner pruningStrategy,
       final Blockchain blockchain,
       final PrunerConfiguration prunerConfiguration) {
-    this(pruningStrategy, blockchain, prunerConfiguration, getDefaultExecutorSupplier());
-  }
-
-  private static Supplier<ExecutorService> getDefaultExecutorSupplier() {
-    return () ->
+    this(
+        pruningStrategy,
+        blockchain,
+        prunerConfiguration,
         Executors.newSingleThreadExecutor(
             new ThreadFactoryBuilder()
                 .setDaemon(true)
                 .setPriority(Thread.MIN_PRIORITY)
                 .setNameFormat("StatePruning-%d")
-                .build());
+                .build()));
   }
 
   public void start() {
-
-    if (state.compareAndSet(State.IDLE, State.RUNNING)) {
-      LOG.info("Starting Pruner.");
-      executorService = executorServiceSupplier.get();
-      pruningStrategy.prepare();
-      blockAddedObserverId = blockchain.observeBlockAdded(this::handleNewBlock);
-    }
+    execute(
+        () -> {
+          if (state.compareAndSet(State.IDLE, State.RUNNING)) {
+            LOG.info("Starting Pruner.");
+            pruningStrategy.prepare();
+            blockAddedObserverId = blockchain.observeBlockAdded(this::handleNewBlock);
+          }
+        });
```

Comment:
Not sure why you can't get your executor and then run the async task?
```suggestion
    if (state.compareAndSet(State.IDLE, State.RUNNING)) {
      LOG.info("Starting Pruner.");
      executorService = executorServiceSupplier.get();
      execute(() -> {
          pruningStrategy.prepare();
          blockAddedObserverId = blockchain.observeBlockAdded(this::handleNewBlock);
      });
    }
```

Reply:
That's how I initially had it 😅. Danno mentioned something about the executor backing up. Tbh I didn't understand his point fully but I was going to look into it later. I was thinking that the AtomicBool should be enough to prevent starting it multiple times.

Reply:
Look at just the first commit in the PR